contrib/babelfishpg_tsql/src/pl_exec.c

```diff
@@ -4693,6 +4693,12 @@ exec_stmt_execsql(PLtsql_execstate *estate,
 			return ret;
 		}
 
+		if (expr->plan && expr->plan->oneshot)
+		{
+			SPI_freeplan(expr->plan);
+			expr->plan = NULL;
+		}
```

Comment:
This code isn't pivot specific anymore so any generic `expr->plan->oneshot` will go here. Has that been tested?

Reply:
1. Yes, it has passed all known cases
2. As for the oneshot variable, is it specific in this case ? My understand is yes, since in PG execution whether oneshot is defined at spi_prepare stage, and it'll release in plan execution ends. But our usage is define at end of execution, and free plan at the begining of query. So they don't have overlap in this case.